### ISSUE TITLE
Support chromecast audio groups

### ISSUE THREAD

**Issue by saulrh** [skorokithakis/catt#119]
[`get_chromecasts` in controllers.py](https://github.com/skorokithakis/catt/blob/master/catt/controllers.py#L32) ensures that all devices can be dialed by creating a new `pychromecast.Chromecast` using the device's `host` parameter. Unfortunately, for audio groups, the `host` param is mostly meaningless and chosen arbitrarily from the devices that participate in the group. This is easily visible from the output of `catt scan`:
```
> catt scan
Scanning Chromecasts...
192.168.1.75 - Bedroom - Google Inc. Google Home
192.168.1.77 - Bedroom TV - Google Inc. Eureka Dongle
192.168.1.70 - Kitchen - Google Inc. Google Home Mini
192.168.1.76 - Living Room - Google Inc. Chromecast Audio
192.168.1.70 - Kitchen - Google Inc. Google Home Mini
```
In this output it's the Kitchen device that's present twice, but this varies across runs as different elements of the group are selected.

Since all device lookups go through `get_chromecasts`, this means that any command directed at an audio group instead operates on a single arbitrary member of the group.

It _appears_ that removing the list comprehension in the final statement of `get_chromecasts` (that is, just return `devices` instead of rebuilding each one with an address) fixes this and lets catt play to chromecast audio groups, but since I don't know why that functionality was added in the first place and was too lazy to get a dev env properly set up, I'll submit a feature request instead of a PR.

**Comment by skorokithakis** [comment]
Thanks for the investigation. Can you clarify a bit why casting to a group will only operate on a single member?

@theychx, what was the reasoning behind the rebuilding? There is a comment about including DIAL info, but I'm not sure why it's necessary.

**Comment by saulrh** [comment]
> Thanks for the investigation. Can you clarify a bit why casting to a group will only operate on a single member?

I'm honestly kind of guessing at what's going on once it gets further up the call stack. To me, it looks like invocations of `setup_cast` will call `get_chromecast` to find a device with a matching name, which then calls `get_chromecasts`, which doesn't return Chromecasts that point at audio groups. If there're code paths that don't go through `setup_cast`, those may work properly. I'm also confused about catt being able to operate on audio groups at _all_, since `get_chromecasts` doesn't yield any devices with the names of an audio group and I'd expect catt to have zero visibility of audio groups. This means I'm probably missing something.

It looks to me like audio groups are implemented as additional chromecast server endpoints hosted on one or more of the devices in the group but on a nonstandard (or at least different) port:
```
In [1]: import pychromecast

In [2]: devices = pychromecast.get_chromecasts()

In [3]: [(cc.host, cc.port, cc.name, cc.cast_type) for cc in devices]
Out[3]:
[('192.168.1.69', 8009, 'Family Room', 'cast'),
 ('192.168.1.75', 8009, 'Bedroom', 'audio'),
 ('192.168.1.77', 8009, 'Bedroom TV', 'cast'),
 ('192.168.1.70', 8009, 'Kitchen', 'cast'),
 ('192.168.1.70', 42019, 'Main Group', 'group'),
 ('192.168.1.76', 8009, 'Living Room', 'audio')]
```
There's at least something more going on in `pychromecast`'s discovery, though:
```
In [5]: [(rebuilt.host, rebuilt.port, rebuilt.name, rebuilt.cast_type) for rebuilt in (pychromecast.Chromecast(host=cc.host, port=cc.port) for cc in devices
   ...: )]
Out[5]:
[('192.168.1.69', 8009, 'Family Room', 'cast'),
 ('192.168.1.75', 8009, 'Bedroom', 'audio'),
 ('192.168.1.77', 8009, 'Bedroom TV', 'cast'),
 ('192.168.1.70', 8009, 'Kitchen', 'cast'),
 ('192.168.1.70', 42019, 'Kitchen', 'cast'),
 ('192.168.1.76', 8009, 'Living Room', 'audio')]

[...]

In [8]: rebuilt_devices = [pychromecast.Chromecast(host=cc.host, port=cc.port) for cc in devices]

In [9]: [(rebuilt.host, rebuilt.port, rebuilt.name, rebuilt.cast_type) for rebuilt in rebuilt_devices]
Out[9]:
[('192.168.1.69', 8009, 'Family Room', 'cast'),
 ('192.168.1.75', 8009, 'Bedroom', 'audio'),
 ('192.168.1.77', 8009, 'Bedroom TV', 'cast'),
 ('192.168.1.70', 8009, 'Kitchen', 'cast'),
 ('192.168.1.70', 42019, 'Kitchen', 'cast'),
 ('192.168.1.76', 8009, 'Living Room', 'audio')]

In [10]: rebuilt_devices[4].media_controller.play_media('http://192.168.1.74:8000/plok_boss.aac', 'audio/aac')
```
This actually does play on the audio group (using a track I publish from my desktop using `python -m http.server`), but the name and cast_type of the device are wrong. It's possible that this an upstream bug in `pychromecast`, or it might just be that I haven't actually read the docs on pychromecast and am going entirely off of `catt`'s codebase and dir. :P

**Comment by theychx** [comment]
@skorokithakis 
> what was the reasoning behind the rebuilding?

When I implemented the cc model checking mechanism, I stumbled upon the odd problem that 1st/2nd gen. Google chromecast devices report different names, depending of whether they are polled via DIAL or zeroconf. So I made the current hack as we use both, and are subsequently using the reported names to determine the appropriate format to pass to ```youtube-dl```.

@saulrh 
The ```Cache``` constructor is calling ```pychromecast.get_chromecasts``` directly because the failing behaviour of our own ```get_chromecasts``` would be a problem when running tests.

It should be possible to solve this. I'll create a branch and work something out.
